core/src/main/java/ai/timefold/solver/core/impl/move/streams/DefaultMoveStreamFactory.java

```diff
@@ -28,19 +27,16 @@ public final class DefaultMoveStreamFactory<Solution_>
 
     private final DataStreamFactory<Solution_> dataStreamFactory;
     private final DatasetSessionFactory<Solution_> datasetSessionFactory;
-    private final ValueRangeManager<Solution_> valueRangeManager;
 
-    public DefaultMoveStreamFactory(SolutionDescriptor<Solution_> solutionDescriptor,
-            ValueRangeManager<Solution_> valueRangeManager) {
-        this.dataStreamFactory = new DataStreamFactory<>(solutionDescriptor, valueRangeManager);
+    public DefaultMoveStreamFactory(SolutionDescriptor<Solution_> solutionDescriptor) {
+        this.dataStreamFactory = new DataStreamFactory<>(solutionDescriptor);
         this.datasetSessionFactory = new DatasetSessionFactory<>(dataStreamFactory);
-        this.valueRangeManager = valueRangeManager;
     }
 
-    public DefaultMoveStreamSession<Solution_> createSession(Solution_ workingSolution, SupplyManager supplyManager) {
-        var session = datasetSessionFactory.buildSession();
-        session.initialize(workingSolution, supplyManager);
-        return new DefaultMoveStreamSession<>(session, workingSolution);
+    public DefaultMoveStreamSession<Solution_> createSession(SessionContext<Solution_> context) {
+        var session = datasetSessionFactory.buildSession(context);
+        session.initialize(context);
+        return new DefaultMoveStreamSession<>(session, context.workingSolution());
```

Comment:
Can we retrieve the working solution from the session instance?

Reply:
That one doesn't store the solution; it doesn't need it for anything.
Therefore I only store the solution in `MoveStreamSession`, where it is actually being used.